lib/ffi-icu/time_formatting.rb

```diff
@@ -96,21 +104,28 @@ def make_formatter(time_style, date_style, locale, time_zone_str, skeleton)
 
     class DateTimeFormatter < BaseFormatter
       def initialize(options={})
-        time_style = options[:time]   || :short
-        date_style = options[:date]   || :short
-        locale     = options[:locale] || 'C'
-        tz_style   = options[:tz_style]
-        time_zone  = options[:zone]
-        skeleton   = options[:skeleton]
-
-        @f = make_formatter(time_style, date_style, locale, time_zone, skeleton)
+        time_style  = options[:time]   || :short
+        date_style  = options[:date]   || :short
+        @locale     = options[:locale] || 'C'
+        tz_style    = options[:tz_style]
+        time_zone   = options[:zone]
+        skeleton    = options[:skeleton]
+        @hour_cycle = options[:hour_cycle] || :pattern
```

Comment:
Why do we need a default here? It's just for short-circuiting correct? Could we just leave it nil?

```ruby
replace_hour_symbol! if @hour_cycle
```